pkg/yurthub/kubernetes/serializer/serializer.go

```diff
@@ -42,24 +42,126 @@ import (
 // YurtHubSerializer is a global serializer manager for yurthub
 var YurtHubSerializer = NewSerializerManager()
 
-// UnsafeDefaultRESTMapper is only used to check whether the GVK is in the scheme according to the GVR information
-var UnsafeDefaultRESTMapper = NewDefaultRESTMapperFromScheme()
+// yurtHubRESTMapperManager is only used in serializer package, for checking resource types
+var yurtHubRESTMapperManager = NewRESTMapperManager(nil)
+
+// RESTMapperManager is responsible for managing different kind of RESTMapper
+type RESTMapperManager struct {
+	sync.RWMutex
+	// UnsafeDefaultRESTMapper is used to save the GVK and GVR mapping relationships of built-in resources
+	unsafeDefaultRESTMapper *meta.DefaultRESTMapper
+	// dynamicRESTMapper is used to save the GVK and GVR mapping relationships of Custom Resources
+	dynamicRESTMapper map[schema.GroupVersionResource]schema.GroupVersionKind
+}
 
-func NewDefaultRESTMapperFromScheme() *meta.DefaultRESTMapper {
+func NewRESTMapperManager(resources map[schema.GroupVersionResource]schema.GroupVersionKind) *RESTMapperManager {
 	s := scheme.Scheme
 	defaultGroupVersions := s.PrioritizedVersionsAllGroups()
 	mapper := meta.NewDefaultRESTMapper(defaultGroupVersions)
 	// enumerate all supported versions, get the kinds, and register with the mapper how to address
 	// our resources.
 	for _, gv := range defaultGroupVersions {
 		for kind := range s.KnownTypes(gv) {
-			//Since RESTMapper is only used for mapping GVR to GVK information,
-			//the scope field is not involved in actual use, so all scope are currently set to meta.RESTScopeNamespace
-			scope := meta.RESTScopeNamespace
-			mapper.Add(gv.WithKind(kind), scope)
+			// Only need to process non-list resources
+			if !strings.HasSuffix(kind, "List") {
+				// Since RESTMapper is only used for mapping GVR to GVK information,
+				// the scope field is not involved in actual use,
+				// so all scope are currently set to meta.RESTScopeNamespace
+				scope := meta.RESTScopeNamespace
+				mapper.Add(gv.WithKind(kind), scope)
+			}
 		}
 	}
-	return mapper
+
+	drm := make(map[schema.GroupVersionResource]schema.GroupVersionKind, len(resources))
+	for gvr, gvk := range resources {
+		drm[gvr] = gvk
+	}
+
+	return &RESTMapperManager{
+		unsafeDefaultRESTMapper: mapper,
+		dynamicRESTMapper:       drm,
+	}
+}
+
+func (rm *RESTMapperManager) IsSchemeResource(gvr schema.GroupVersionResource) (bool, schema.GroupVersionKind) {
```

Comment:
I think it's not friendly to expose `RESTMapperManager` out of serializer.go, so we need to delete `IsSchemeResource` and `IsCustomResource` func, but add `KindFor` func for `Serializer` instead.

Reply:
Thanks for the suggestion, the code has been updated.